### PR TITLE
[NOD-1148] P2P stabilization

### PR DESCRIPTION
**Pull request by svarogg**

### CODE REVIEW

File: netadapter/router/route.go [hunk 1]
```diff
@@ -47,16 +54,17 @@ func (r *Route) Enqueue(message wire.Message) (isOpen bool) {
...
 }
 
 // EnqueueWithTimeout attempts to enqueue a message to the Route
 // and returns an error if the given timeout expires first.
 func (r *Route) EnqueueWithTimeout(message wire.Message, timeout time.Duration) (isOpen bool, err error) {
+	r.closeLock.Lock()
+	defer r.closeLock.Unlock()
+
 	if r.closed {
+		fmt.Printf("OMG IM CLOSED!!!")
```
Comment (svarogg) [inline netadapter/router/route.go:67]: Please remove

File: netadapter/router/route.go [hunk 2]
```diff
@@ -47,16 +54,17 @@ func (r *Route) Enqueue(message wire.Message) (isOpen bool) {
 
 // Dequeue dequeues a message from the Route
 func (r *Route) Dequeue() (message wire.Message, isOpen bool) {
```
Comment (stasatdaglabs) [inline netadapter/router/route.go:56]: Please lock closedLock here.

File: netadapter/router/route.go [hunk 3]
```diff
@@ -73,14 +81,11 @@ func (r *Route) EnqueueWithTimeout(message wire.Message, timeout time.Duration)
 // DequeueWithTimeout attempts to dequeue a message from the Route
 // and returns an error if the given timeout expires first.
 func (r *Route) DequeueWithTimeout(timeout time.Duration) (message wire.Message, isOpen bool, err error) {
```
Comment (stasatdaglabs) [inline netadapter/router/route.go:83]: Please lock closedLock here.

File: netadapter/router/route.go [hunk 4]
```diff
@@ -70,17 +73,20 @@ func (r *Route) EnqueueWithTimeout(message wire.Message, timeout time.Duration)
 	}
 }
 
+// Dequeue dequeues a message from the Route
+func (r *Route) Dequeue() (message wire.Message, isOpen bool) {
+	message, isOpen = <-r.channel
+	return
```
Comment (stasatdaglabs) [inline netadapter/router/route.go:79]: Please defancify this statement.

**Pull request merged**